api/bazel/repositories.bzl

```diff
@@ -34,6 +34,16 @@ def api_dependencies():
         locations = REPOSITORY_LOCATIONS,
         build_file_content = KAFKASOURCE_BUILD_CONTENT,
     )
+    envoy_http_archive(
+        name = "kafka_server_binary",
```

Comment:
Sorry, no context on this review, but are we bringing in external binaries as part of these build changes? Can you elaborate here on what these two new deps are?

Reply:
I want to run a real integration test (with kafka server, envoy, and (python) kafka client), so to achieve that I need the kafka server binary & python kafka client dependency.

the test itself https://github.com/envoyproxy/envoy/pull/8188/files#diff-b970f74ad68d76aca06e396c34844402R242

Reply:
Do we have a sense for how long these new tests take in the common case? I'm mostly just worried that we might greatly increase CI time, though I think it's really awesome to have a self contained real integration test. 

Also, have you run these tests through a `--runs_per_test=1000` pass to check for flakes? I would be very worried about flakes here.

Reply:
On my mac, it's usually <= 60 seconds.
On the build machine, I've seen the values around 120 seconds, with the highest 240s in tsan build in https://dev.azure.com/cncf/envoy/_build/results?buildId=21508

The 1000-runs succeeded (with 8 tests running in parallel):
```
bazel test //test/extensions/filters/network/kafka/broker/integration_test:kafka_broker_integration_test --runs_per_test 1000

INFO: 2000 processes: 2000 darwin-sandbox.
INFO: Build completed successfully, 1002 total actions
//test/extensions/filters/network/kafka/broker/integration_test:kafka_broker_integration_test PASSED in 58.1s
  Stats over 1000 runs: max = 58.1s, min = 43.9s, avg = 50.7s, dev = 2.5s
```